tests/bfd/test_bfd_static_route.py

```diff
@@ -145,17 +147,12 @@ def test_bfd_with_lc_reboot_ipv4(
         # Waiting for all processes on Source dut
         wait_critical_processes(src_dut)
 
-        check_bgp = request.getfixturevalue("check_bgp")
-        results = check_bgp()
-        failed = [
-            result for result in results if "failed" in result and result["failed"]
-        ]
-        if failed:
-            pytest.fail(
-                "BGP check failed, not all BGP sessions are up. Failed: {}".format(
-                    failed
-                )
-            )
+        assert wait_until(
+            300,
+            10,
+            0,
+            lambda: check_bgp_status(request),
+        )
```

Comment:
I found that we need more time to start up BGP sessions, especially in the tests where make changes to RP, so I changed all BGP checks in this test file to a timeout function. Non-RP related tests will have a timeout of 300s, while RP related tests will have a timeout of 600s